### ISSUE TITLE
[Feature Request]: add a "sync this page" button

### ISSUE THREAD

**Issue by INPoppoRTUNE** [remotely-save/remotely-save#607]
### What feature are you suggesting?

I use Obsidian on multiple devices and my vault is getting quite big. As logical consequence the syncing periods are becoming longer and longer as time goes on.
I often find myself in the condition of having an image on my pc that I want to annotate using my iPad's pen: so I add the image to the relative note, wait several minutes to sync the entire vault from pc to cloud and from cloud to iPad, annotate the image with the pen and re-sync the vault once again.

A "sync this page" button would allow me to cut several minutes from this process.

Thanks for your time and for the wonderful plugin!

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
a good idea, but is it possible that users might forget to click the button...

**Comment by INPoppoRTUNE** [comment]
> a good idea, but is it possible that users might forget to click the button...

I'd like to have this function as a manual trigger to run in parallel with the scheduled autorun, not as alternative.

**Comment by dawnlh** [comment]
The large vault does impose great pressure on mobile devices' memory. It will be convenient to sync (pull) required files instead all of them on the mobile device. For example, when you open a document in your mobile device, only the file itself and the related files (images&refered files) will be pulled/downloaded locally.

**Comment by dprbook** [comment]
> The large vault does impose great pressure on mobile devices' memory. It will be convenient to sync (pull) required files instead all of them on the mobile device. For example, when you open a document in your mobile device, only the file itself and the related files (images&refered files) will be pulled/downloaded locally.

True, that would save on memory, but goes against Obsidian's philosophy, I think. 
Check out [this post on the Obsidian Forum](https://forum.obsidian.md/t/the-future-is-local-and-centralized/5333).
Also, while rclone (which this plugin uses) can do that, it would require quite a bit of rework to track individual files instead of just "sync the whole folder". 
When do you remove the local files? Be careful to not delete the file on the remote then as well. Then, you need an option to delete the file for real, which requires hooking into Obsidian's file handler or such.
Quite a bit of work for a community plugin, if you ask me.

My solution was to split my vault up into smaller ones by domain and share the config(s) across all vaults using sym links, but that has other issues when it comes to syncing (outlined here https://github.com/remotely-save/remotely-save/issues/835).

**Comment by controlledentropy** [comment]
Beautiful idea, the best sync plan is no sync plan. Just modify one then click sync button even auto sync current file like what OneNote does.

This function(or button) should not be so difficult for the great devs of this plugin😁.

Light, frequently, and efficiently
